### PR TITLE
fixes #13182 - add support for puppet distributor

### PR DESCRIPTION
**Pull request by bbuckingham**
This commit updates puppet repositories to support the pulp
'puppet distributor', in addition to the existing 'puppet install distributor'.

### CODE REVIEW

File: app/lib/actions/pulp/repository/create.rb [hunk 1]
```diff
@@ -104,6 +105,14 @@ def iso_distributor
           end
         end
 
+        def puppet_distributor
+          options = { id: "#{input[:pulp_id]}_puppet", auto_publish: true }
+          Runcible::Models::PuppetDistributor.new(input[:path],
```
Comment (jlsherrill) [inline app/lib/actions/pulp/repository/create.rb]: I *think* input[:path] here is pointing to the puppet env in /etc/puppet/environments/SOMETHING  which may not make sense for the normal distributor.  It all seems to work, as it seems to be published in:

/var/lib/pulp/published/puppet/https/repos/Default_Organization-Library-asdfasdf/

properly, but i'm not fully sure what that option actually does.

Comment (jlsherrill) [inline app/lib/actions/pulp/repository/create.rb]: oddly i don't even see this option (absolute_path) in the pulp-admin puppet repo create command.  Maybe that is just left over in runcible from a long time ago and isn't used anymore?  Might ask the pulp team

**Comment by jlsherrill** [comment]
Uploading puppet modules doesn't seem to be kicking off a metadata generate, probably due to this line:  https://github.com/Katello/katello/blob/master/app/lib/actions/katello/repository/finish_upload.rb#L6

as previously we didn't have anything to generate, but now we do

**Comment by jlsherrill** [comment]
Also, could we change this part of the repo details page to show the repo URL and allow the user to change if its published via http:

https://github.com/Katello/katello/blob/master/engines/bastion_katello/app/assets/javascripts/bastion_katello/repositories/details/views/repository-info.html#L143-L155

**Comment by jlsherrill** [comment]
Everything else looks great :+1:

**Comment by bbuckingham** [comment]
@jlsherrill, Updated.  Thanks for the comments!

Note: I updated the repo new & info to allow the user to set the 'unprotected' field.

**Comment by jlsherrill** [comment]
Opened http://projects.theforeman.org/issues/13365 to add some options in the future, but ACK from me

**Pull request merged**